src/Conversion/ONNXToKrnl/Tensor/ConstantOfShape.cpp

```diff
@@ -75,21 +75,22 @@ struct ONNXConstantOfShapeOpLowering : public ConversionPattern {
     } else
       llvm_unreachable("unsupported element type");
 
-    SmallVector<Value, 4> loopIVs;
+    KrnlBuilder createKrnl(rewriter, loc);
     // Create a Krnl iterate if the output is not a scalar tensor.
     if (!hasAllScalarValues({alloc})) {
-      BuildKrnlLoop loops(rewriter, loc, rank);
-      loops.createDefineAndIterateOp(alloc);
-      Block *iterationBlock = loops.getIterateBlock();
-      // Get IVs.
-      for (auto arg : iterationBlock->getArguments())
-        loopIVs.push_back(arg);
-      // Insert instructions inside the KernelIterateOp body.
-      rewriter.setInsertionPointToStart(iterationBlock);
-    }
-
-    // Store the constant value to the output.
-    create.krnl.store(constantVal, alloc, loopIVs);
+      IndexExprScope childScope(&rewriter, loc);
+      ValueRange loopDef = createKrnl.defineLoops(rank);
+      SmallVector<IndexExpr, 4> lbs(rank, LiteralIndexExpr(0));
+      MemRefBoundsIndexCapture allocBounds(alloc);
+      SmallVector<IndexExpr, 4> ubs;
+      allocBounds.getDimList(ubs);
+      createKrnl.iterateIE(loopDef, loopDef, lbs, ubs,
+          [&](KrnlBuilder &createKrnl, ValueRange loopInd) {
+            // Store the constant value to the output.
```

Comment:
[nit]: the comment does not add anything useful, is obvious a store instruction is generated to store the constant value to the output from the code itself. Please remove.